### PR TITLE
Proj inbound calling

### PR DESCRIPTION
**Pull request by alonso-cadenas**
## Description

Jira Ticket: [CALL-5895](https://issues.hubspotcentral.com/browse/CALL-5895)

This PR aims at adding support for following new features. 

- inbound calling feature
- caller id match on incoming call 
- navigate to record if caller id match is found
- availability status change
- Inbound calling support in demo apps

## Merge Checklist

| Q                        | A 
| ------------------------ | --------------
| Adds Documentation?      |
| Any Dependency Changes?  |
| Patch: Bug Fix?          |
| Minor: New Feature?      | ✅ 
| Major: Breaking Change?  |

[BRAVE Checklist](https://github.com/HubSpot/calling-extensions-sdk/blob/master/SHIP_WITH_CARE.md)

- [x] I have read the BRAVE checklist and confirmed if the following is necessary.

| Q                              | A 
| ------------------------------ | --------------
| Backwards Compatible?          | ✅ 
| Rollout/Rollback Plan?         | 
| Automated test coverage?       | 
| Verified that changes work?    | ✅ 
| Expect Dependencies to Fail?   |

### CODE REVIEW

**Review by hemang-thakkar — APPROVED** [review]
LGTM

File: .github/workflows/preview.yml [hunk 1]
```diff
@@ -8,7 +8,8 @@ on:
       - reopened
       - synchronize
     paths:
-      - 'demos/**'
+      - "demos/**"
+  workflow_dispatch:
```
Comment (hemang-thakkar) [inline .github/workflows/preview.yml:12]: is this an empty tag??

File: demos/demo-react-ts/src/components/screens/CallEndedScreen.tsx [hunk 2]
```diff
@@ -41,7 +44,13 @@ function CallEndedScreen({
   return (
     <Wrapper>
       <div style={{ textAlign: "center" }}>
-        <h2>Call with {dialNumber} ended</h2>
+        <h2>
+          Call with{" "}
+          {direction === "OUTBOUND"
+            ? formatPhoneNumber(dialNumber)
+            : formatPhoneNumber(incomingNumber)}{" "}
```
[the remark this answers is not in the document]
Comment (hemang-thakkar) [reply demos/demo-react-ts/src/components/screens/CallEndedScreen.tsx:51]: it looks like "potato Potaato". am i missing anything?

**Review by esme — APPROVED** [review]

File: demos/demo-minimal-js/index.html [hunk 3]
```diff
@@ -120,7 +124,29 @@
...
+          id="userunavailable"
+          onclick="userUnavailable()"
+          type="button"
+          value="user unavailable"
+        />
+        <p>4. Start a call</p>
+        <input
+          disabled
+          id="incomingcall"
+          onclick="incomingCall()"
```
[the remark this answers is not in the document]
Comment (alonso-cadenas) [reply demos/demo-minimal-js/index.html:191]: @hemang-thakkar, [one last request](https://github.com/HubSpot/calling-extensions-sdk/pull/179#discussion_r1511448014) regarding this topic and I agree that we should be good to merge this PR.

**Comment by alonso-cadenas** [comment]
Good to merge!

**Pull request merged**